sklearn/tree/_criterion.pyx

```diff
@@ -1438,7 +1442,12 @@ cdef class Poisson(RegressionCriterion):
         for k in range(n_outputs):
             y_mean = y_sum[k] / weight_sum
 
-            if y_mean <= 0:
+            if y_mean <= EPSILON:
```

Comment:
In this case, I would change to `y_sum[k] <= EPSILON` as `weight_sum` can be very large.

Reply:
Oh right